swag/db.py

```diff
@@ -85,24 +108,64 @@ def add_user(self, user, guild):
         else:
             raise AccountAlreadyExist
 
+    def add_cagnotte(self, cagnotte_name, cagnotte_currency, cagnotte_creator_id):
+        if cagnotte_name not in self.cagnotte_id:
+            self.cagnotte_id[cagnotte_name] = self.next_cagnotte_id
+            self.next_cagnotte_id += 1
+
+            self.cagnotte_name.append(cagnotte_name)
+            if cagnotte_currency == Currency.SWAG:
+                self.cagnotte_balance.append(0)
+            elif cagnotte_currency == Currency.STYLE:
+                self.cagnotte_balance.append(Decimal(0))
+
+            self.cagnotte_currency.append(cagnotte_currency)
+            self.cagnotte_manager.append([cagnotte_creator_id])
+            self.cagnotte_participant.append(set())
+            self.cagnotte_activation.append(True)
+
+        else:
+            raise CagnotteNameAlreadyExist
+
     def get_account(self, user):
         try:
             return SwagAccount(self, self.id[user])
         except KeyError:
             raise NoAccountRegistered(user)
 
+    def get_cagnotte(self, cagnotte_name):
+        try:
+            return Cagnotte(self, self.cagnotte_id[cagnotte_name])
+        except KeyError:
+            raise NoCagnotteRegistered(cagnotte_name)
+
     def get_account_from_index(self, idx):
         return SwagAccount(self, idx)
 
+    def get_cagnotte_from_index(self, idx):
+        return Cagnotte(self, idx)
+
     def get_accounts(self):
         return (SwagAccount(self, idx) for idx in range(self.user_number()))
 
+    def get_cagnottes(self):
+        return (Cagnotte(self, idx) for idx in range(self.cagnotte_number()))
+
     def get_account_infos(self):
         return (AccountInfo(account) for account in self.get_accounts())
 
+    def get_cagnotte_infos(self):
+        return (CagnotteInfo(cagnotte) for cagnotte in self.get_cagnottes())
+
+    def get_active_cagnotte_ids(self):
```

Comment:
Where is this needed?

Reply:
in get_active_cagnotte function in bank.py

Reply:
Then I suggest we remove this function in favor of the snipped I'm going to put in comment for `get_active_cagnotte`

Reply:
Resolved here : https://github.com/BobbySoftwares/Bybbot/pull/7/commits/4417c76019e6129a0b424d6131cdc41bc81da069